ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/ShortcutEditDialogFragment.java

```diff
@@ -131,10 +136,24 @@ public void afterTextChanged(Editable editable) {
                         Toast.makeText(getActivity(), R.string.error_invalid_icon_format, Toast.LENGTH_LONG).show();
                     }
 
+                    final String action;
+                    if (binding.checkBoxAction.isChecked()) {
+                        action = binding.editTextAction.getText().toString();
+                    } else {
+                        action = null;
+                    }
+
+                    final Uri data;
+                    if (binding.checkBoxData.isChecked()) {
+                        data = Uri.parse(binding.editTextData.getText().toString());
```

Comment:
Handle parsing error - optimally already indicate in input component and don't allow submitting